lib/ui/namespaces.js

```diff
@@ -32,6 +32,114 @@ function namespaces_list(screen) {
     }
   });
 
+  const search = blessed.textbox({
+    parent       : namespaces_list,
+    label        : 'Search',
+    content      : '',
+    border       : 'line',
+    width        : '30%',
+    height       : 3,
+    right        : 2,
+    top          : -1,
+    keys         : true,
+    mouse        : true,
+    hidden       : true,
+    inputOnFocus : true,
+    style : {
+      fg       : 'white',
+      label    : { bold: true },
+      border   : { fg: 'white' },
+      selected : { bg: 'blue' },
+    }
+  });
+
+  search.__oolistener = search._listener;
```

Comment:
I would propose to add a comment that this is the canonical way of extending component in Blessed.